website/docs/r/iothub.html.markdown

```diff
@@ -294,6 +296,14 @@ A `fallback_route` block supports the following:
 
 A `file_upload` block supports the following:
 
+* `authentication_type` - (Optional) Type used to authenticate against the storage account. Possible values are `keyBased` and `identityBased`. Defaults to `keyBased`.
+
+* `identity_id` - (Optional) ID of the User Managed Identity used to authenticate against the storage account.
+
+-> **NOTE:** `identity_id` can only be specified when `authentication_type` is `identityBased`. It must be one of the `identity_ids` of the Iot Hub. If not specified when `authentication_type` is `identityBased`, System Assigned Managed Identity of the Iot Hub will be used.
+
+~> **NOTE:** System Assigned Managed Identity can only be used in an update because access to the endpoint cannot be granted before the creation is done. The extracted resources `azurerm_iothub_file_upload` can be used to configure File Upload with System Assigned Managed Identity without the need of an update.
```

Comment:
This might be better phrased as
```suggestion
~> **NOTE:** An IoT Hub can only be updated to use the System Assigned Managed Identity for `file_upload` since it is not possible to grant access to the endpoint until after creation. The extracted resource `azurerm_iothub_file_upload` can be used to configure File Upload with the IoT Hub's System Assigned Managed Identity without the need for an update.
```